src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -390,7 +390,11 @@ private void AddDirectoryWatchUnlocked(WatchedDirectory? parent, string director
                     // of the world, but there's little that can be done about that.)
                     if (directoryEntry.Parent != parent)
                     {
-                        directoryEntry.Parent?.Children!.Remove (directoryEntry);
+                        if (directoryEntry.Parent != null)
```

Comment:
```suggestion
                        // Work around https://github.com/dotnet/csharplang/issues/3393 preventing Parent?.Children!. from behaving as expected
                        if (directoryEntry.Parent != null)
```

Reply:
I initially thought to add it but then I thought that we introduced it while annotating by doing tiny refactoring so wasn't sure about adding it. Will change